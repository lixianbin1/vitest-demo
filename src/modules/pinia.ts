import { App } from 'vue'
import { createPinia } from 'pinia'

export const PiniaStore = createPinia()
export const install = (app:App) => {
  app.use(PiniaStore)
}
export default install