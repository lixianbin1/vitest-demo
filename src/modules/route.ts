import { App } from 'vue'
import { createRouter,createWebHashHistory } from 'vue-router'
import { setupLayouts } from 'virtual:generated-layouts'
import generatedRoutes from '~pages'

export const routes = setupLayouts(generatedRoutes)
export const router = createRouter({
  history: createWebHashHistory(),
  routes,
})
export const install = (app:App) => {
  app.use(router)
}
export default install