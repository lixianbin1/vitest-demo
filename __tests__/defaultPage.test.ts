import {expect,test,vi} from 'vitest'
import {render} from '@testing-library/vue'
import defaultPage from '../src/pages/[...all].vue'
import {i18n} from '../src/modules/i18n'
import {router} from '../src/modules/route'

test('defaultPage页面的快照',()=>{
  const { container } = render(defaultPage,{
    global: {
      plugins:[i18n,router]
    }
  })
  expect(container).toMatchSnapshot()
})