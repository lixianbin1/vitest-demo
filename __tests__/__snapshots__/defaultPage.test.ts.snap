// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`defaultPage页面的快照 1`] = `
<div>
  <main
    flex=""
    items-center=""
    justify-center=""
    style="height: 100%; background: rgb(204, 204, 204);"
  >
    <div
      class="el-card is-always-shadow"
      style="text-align: center; padding-bottom: 20px; margin-bottom: 100px; position: relative; border-radius: 20px;"
      w-450px=""
    >
      <!--v-if-->
      <div
        class="el-card__body"
        style=""
      >
        
        <h1
          class="animation"
          flex=""
          items-center=""
          justify-center=""
          style="letter-spacing: 30px; padding: 30px 0px 10px 15px; font-weight: bold;"
          text-100px=""
        >
           404 
        </h1>
        <button
          bth=""
          class="backbutton"
        >
           Back 
        </button>
        
      </div>
      <!--v-if-->
    </div>
  </main>
</div>
`;
